src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Models/ExtractToComponentCodeActionParams.cs

```diff
@@ -14,5 +14,5 @@ internal sealed class ExtractToComponentCodeActionParams
     public int End { get; set; }
 
     [JsonPropertyName("namespace")]
-    public required string Namespace { get; set; }
+    public required string? Namespace { get; set; }
```

Comment:
>required [](http://example.com/codeflow?start=11&length=8)

required usage here confuses me. I would have thought the position parameters would have been required and the namespace wouldn't be

Reply:
Ahh yeah, the required can go, it's a convenient trick for DTOs with non-nullable types to make the compiler not complain that you haven't specified an initial value of the field. Not needed now that it's nullable obviously.

You're also right that it would make sense to be on the position parameters. In general there are a few things that annoy me about our code actions code, and having so many bits in separate files is one of them. These params types should probably all just be single line records defined inline in a combined provider-and-resolver. One day when I'm bored on the couch.